services/121-service/src/payments/fsp-integration/excel/excel.service.ts

```diff
@@ -111,11 +111,15 @@ export class ExcelService {
       );
     }
     // This method joins the registrations and transactions arrays based on the referenceId.
-    // Both arrays are assumed to be sorted by referenceId. This allows us to use a two-pointer
+    // We sort both arrays by referenceId. This allows us to use a two-pointer
     // technique to join the arrays, which is more efficient than using a nested loop or the find method.
     const transactionsOrdered = transactions.sort((a, b) =>
       a.registration.referenceId.localeCompare(b.registration.referenceId),
     );
+    // Make sure registrations are also ordered by referenceId
+    orderedRegistrations.sort((a, b) =>
```

Comment:
@nielsbom the naming is off a bit then now, because the input of this method is already called 'orderedRegistrations'. You should either do the ordering earlier (before this method), or change the naming.